core/src/main/java/com/linecorp/armeria/common/metric/DefaultMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,141 @@
+        checkArgument(completeRequestPrefixSize > 0, "completeRequestPrefixSize: %s (expected: > 0)",
+                      completeRequestPrefixSize);
+        this.activeRequestPrefixSize = activeRequestPrefixSize;
+        this.completeRequestPrefixSize = completeRequestPrefixSize;
+    }
+
+    @Override
+    public MeterIdPrefix activeRequestPrefix(MeterRegistry registry, RequestOnlyLog log) {
+        final Builder<Tag> tagListBuilder =
+                ImmutableList.builderWithExpectedSize(activeRequestPrefixSize);
+        addActiveRequestPrefixTags(tagListBuilder, log);
+        return new MeterIdPrefix(name, tagListBuilder.build());
+    }
+
+    @Override
+    public MeterIdPrefix completeRequestPrefix(MeterRegistry registry, RequestLog log) {
+        final Builder<Tag> tagListBuilder =
+                ImmutableList.builderWithExpectedSize(completeRequestPrefixSize);
+        addCompleteRequestPrefixTags(tagListBuilder, log);
+        return new MeterIdPrefix(name, tagListBuilder.build());
+    }
+
+    /**
+     * Adds the active request tags in lexicographical order for better sort performance.
+     */
+    protected void addActiveRequestPrefixTags(
+            Builder<Tag> tagListBuilder, RequestOnlyLog log) {
+        addHostnamePattern(tagListBuilder, log);
+        addMethodAndService(tagListBuilder, log);
+    }
+
+    /**
+     * Adds the complete request tags in lexicographical order for better sort performance.
+     */
+    protected void addCompleteRequestPrefixTags(Builder<Tag> tagListBuilder, RequestLog log) {
+        addHostnamePattern(tagListBuilder, log);
+        addHttpStatus(tagListBuilder, log);
+        addMethodAndService(tagListBuilder, log);
+    }
+
+    private static void addHostnamePattern(Builder<Tag> tagListBuilder, RequestOnlyLog log) {
+        final RequestContext ctx = log.context();
+        if (ctx instanceof ServiceRequestContext) {
+            final ServiceRequestContext sCtx = (ServiceRequestContext) ctx;
+            tagListBuilder.add(Tag.of(Flags.useLegacyMeterNames() ? "hostnamePattern"
+                                                                  : "hostname.pattern",
+                                      sCtx.config().virtualHost().hostnamePattern()));
+        }
+    }
+
+    private static void addMethodAndService(Builder<Tag> tagListBuilder, RequestOnlyLog log) {
+        String methodName = log.name();
+        if (methodName == null) {
+            final RequestHeaders requestHeaders = log.requestHeaders();
+            methodName = requestHeaders.method().name();
+        }
+        tagListBuilder.add(Tag.of("method", methodName));
+        final String serviceName = log.serviceName();
+        if (serviceName != null) {
+            tagListBuilder.add(Tag.of("service", serviceName));
+        }
+    }
+
+    /**
+     * Adds {@link HttpStatus} to {@link Tag}.
+     */
+    protected static void addHttpStatus(Builder<Tag> tagListBuilder, RequestLog log) {
```

Comment:
Haven't been following the changes to the default id prefixes - don't these have too much cardinality to add by default?

Reply:
We were actually adding the HTTP status by default.
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunction.java#L85
And I found an issued that you created. https://github.com/line/armeria/issues/1981
😆

Reply:
Perhaps we can add grpc.status only for gRPC metrics. 🤔

Reply:
> Perhaps we can add grpc.status only for gRPC metrics. 🤔 

If you meant 'no `http.status` for gRPC metrics, I guess not because they have meaning at different levels.

Reply:
Yeah I guess I thought a http status could sort of DOS because of too many statuses, even when health check has less variety. But most servers won't have much variety in normal http status too so it should be ok.